File: SCTE65/src/java/other/org/himalay/si/transport_protocol_descriptor.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.si ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Sun Sep 14 22:48:36 EDT 2014")

public  class transport_protocol_descriptor extends   app_desc_baseFactory.app_desc_base { //Concrete type is transport_protocol_descriptor




    // members variables
    // desc_header
    public descriptor_header desc_header ;
    // id
    public int id ;
    // label
    public short label ;
    // url
    public ByteArray url ;

        public transport_protocol_descriptor () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize desc_header
        desc_header = new descriptor_header();
        // Initialize id
        
        // Initialize label
        
        // Initialize url
        url= new ByteArray();url.setSizeType("FIRST_UI8");
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
  preRead();
    int retVal= 0;
    DataInputStream saved = istream;
    int sizeLimit= getHeader().length + (0);
    if ( sizeLimit > 0 ){
    byte[] ba        = new byte[sizeLimit];
    istream.readFully(ba);
    ByteArrayInputStream baTemp= new ByteArrayInputStream(ba);
    istream = new DataInputStream(baTemp);
    {
                    // read id
        {id=istream.readUnsignedShort(); retVal+=2;}
        // read label
        {label=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read url
        {retVal+=url.read(istream); }

}
}
istream = saved;
postRead();
        return retVal;
        }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

         DataInputStream saved = istream;
    int sizeLimit= getHeader().length + (0);
    if ( sizeLimit > 0 ){
    byte[] ba        = new byte[sizeLimit];
    istream.readFully(ba);
    ByteArrayInputStream baTemp= new ByteArrayInputStream(ba);
    istream = new DataInputStream(baTemp);
    {
                    // read desc_header
        retVal += desc_header.read(istream);
        // read id
        {id=istream.readUnsignedShort(); retVal+=2;}
        // read label
        {label=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read url
        {retVal+=url.read(istream); }

                }
}
istream = saved;
postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        { /** fix dependent sizes for desc_header **/  }
        
        
        { /** fix dependent sizes for url**/  }
    
                // write desc_header
        if (desc_header!=null)retVal +=desc_header.write(ostream);
        // write id
        ostream.writeShort(id); retVal +=2;
        // write label
        ostream.writeByte(label); retVal +=1;
        // write url
        {retVal += url.write(ostream);}
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("transport_protocol_descriptor\n");
    dc.increaseIndent();
        int retVal= 0;
        // write desc_header
        if ( desc_header != null ) {dc.indent();dc.getPs().println("desc_header") ;retVal +=desc_header.dump(dc);}
        // write id
        dc.indent();dc.getPs().println("id="+id+"(0x"+ Integer.toHexString(id) + ")" );
        // write label
        dc.indent();dc.getPs().println("label="+label+"(0x"+ Integer.toHexString(label) + ")" );
        // write url
        dc.indent();dc.getPs().print("url: "+url.getSize()+"(0x"+Integer.toHexString(url.getSize())+")\n");this.url.dump(dc);
dc.decreaseIndent();
        return retVal;
    }


        // Getter for desc_header
    //public descriptor_header getDesc_header()
    //{
    //    return desc_header ;
    //}

    
    // Setter for desc_header
    //public void setDesc_header(descriptor_header val)
    //{
    //    this.desc_header= val;
    //}
    // Getter for id
    //public int getId()
    //{
    //    return id ;
    //}

    
    // Setter for id
    //public void setId(int val)
    //{
    //    this.id= val;
    //}
    // Getter for label
    //public short getLabel()
    //{
    //    return label ;
    //}

    
    // Setter for label
    //public void setLabel(short val)
    //{
    //    this.label= val;
    //}
    // Getter for url
    //public ByteArray getUrl()
    //{
    //    return url ;
    //}

    
    // Setter for url
    //public void setUrl(ByteArray val)
    //{
    //    this.url= val;
    //}


    public void setUrl(byte[] val)
    {
        this.url.setData(val);
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        
     public void setHeader(descriptor_header header)
    {
         this.desc_header= header;   
    }
    
    public descriptor_header getHeader()
    {
        return this.desc_header;
    }

}

// End of code

File: libsi/src/bp/transport_protocol_descriptor.cpp
//**//
//@Created(date = "Mon Feb 01 22:29:59 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "transport_protocol_descriptor.h"

#ifdef org_himalay_si_transport_protocol_descriptor__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace si {

using namespace ::org::himalay::msgs::runtime;

// Constructor
transport_protocol_descriptor::transport_protocol_descriptor() {
    references= 0;


    // desc_header
    org_himalay_si_descriptor_header__NEW( desc_header);
    // id
    id= (ui16)0;
    // label
    label= (ui8)0;
    // url
    org_himalay_msgs_runtime_ByteArray_NEW(url);
    url->setSizeType("FIRST_UI8");
}
// Destructor
transport_protocol_descriptor::~transport_protocol_descriptor() {
    // desc_header
    org_himalay_si_descriptor_header__DELETE( desc_header);
    // id

    // label

    // url
    org_himalay_msgs_runtime_ByteArray_DELETE( url);
}

int transport_protocol_descriptor::readNoHeader(DataInputStream&  distream ) { // throws IOException
    int retVal = 0;
    int sizeLimit= desc_header->length + (0);
    ui8* ba        = new ui8[sizeLimit];
    distream.readFully(ba,0,sizeLimit);
    ByteArrayDataInputStream istream(ba,0, sizeLimit);
    {

        // id
        {
            id=istream.readUI16();
            retVal+=2;
        }
        // label
        {
            label=(istream.readUI8());
            retVal+=1;
        }
        // url
        {
            retVal+=url->read(istream);
        }
    }
    return retVal;
}

int transport_protocol_descriptor::read(DataInputStream&  distream ) { // throws IOException
    int retVal = 0;
    int sizeLimit= desc_header->length + (0);
    ui8* ba        = new ui8[sizeLimit];
    distream.readFully(ba, 0, sizeLimit);
    ByteArrayDataInputStream istream(ba, 0, sizeLimit);
    {
        // read desc_header
        retVal += desc_header->read(istream);
        // read id
        {
            id=istream.readUI16();
            retVal+=2;
        }
        // read label
        {
            label=(istream.readUI8());
            retVal+=1;
        }
        // read url
        {
            retVal+=url->read(istream);
        }
    }

    return retVal;
}

int transport_protocol_descriptor::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for desc_header **/
    }


    {   /** fix dependent sizes for url**/
    }

    int retVal= 0;
    // write desc_header
    if ( desc_header != NULL ) {
        retVal +=desc_header->write(ostream);
    }
    // write id
    ostream.writeUI16(id);
    retVal +=2;
    // write label
    ostream.writeUI8(label);
    retVal +=1;
    // write url
    {
        retVal += url->write(ostream);
    }

    return retVal;
}

int transport_protocol_descriptor::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("transport_protocol_descriptor\n");
    dc.increaseIndent();
    int retVal= 0;
    // write desc_header
    if ( desc_header != NULL ) {
        dc.indent();
        dc.getPs().println("desc_header") ;
        retVal +=desc_header->dump(dc);
    }
    // write id
    dc.indent();
    dc.getPs().print("id=");
    dc.getPs().println((long)id);
    // write label
    dc.indent();
    dc.getPs().print("label=");
    dc.getPs().println((long)label);
    // write url
    dc.indent();
    dc.getPs().print("url");
    url->dump(dc);
    dc.decreaseIndent();
    return retVal;
}


// Getter for desc_header
//org_himalay_si_descriptor_header__PTR_TYPE transport_protocol_descriptor::getDesc_header()
//{
//return this->desc_header;
//}

// Setter for desc_header
//void transport_protocol_descriptor::setDesc_header(org_himalay_si_descriptor_header__PTR_TYPE val)
//{
//this->desc_header= val;
//}

// Getter for id
//ui16 transport_protocol_descriptor::getId()
//{
//return this->id;
//}

// Setter for id
//void transport_protocol_descriptor::setId(ui16 val)
//{
//this->id= val;
//}

// Getter for label
//ui8 transport_protocol_descriptor::getLabel()
//{
//return this->label;
//}

// Setter for label
//void transport_protocol_descriptor::setLabel(ui8 val)
//{
//this->label= val;
//}

// Getter for url
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE transport_protocol_descriptor::getUrl()
//{
//return this->url;
//}

// Setter for url
//void transport_protocol_descriptor::setUrl(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->url= val;
//}

int transport_protocol_descriptor::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

void transport_protocol_descriptor::setHeader(org_himalay_si_descriptor_header__PTR_TYPE header)
{
    this->desc_header= header;
}

org_himalay_si_descriptor_header__PTR_TYPE transport_protocol_descriptor::getHeader()
{
    return this->desc_header;
}

#ifdef org_himalay_si_transport_protocol_descriptor__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::si::transport_protocol_descriptor* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::si::transport_protocol_descriptor* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code